### PR TITLE
14 create chart class and postprocessing interface

### PR DESCRIPTION
**Pull request by matdenkas**
I created two interfaces, one for ingestion, one for post processing. As we are using data frames this time I want to try and use a system similar to the collection of series system we use at the end of semaphore. So I designed this in such a way that instead of using explicit data series, we use abstract calls to classes and a single DataFrame that is collating all the data. 

Nothing really to test yet but please take a look!

### CODE REVIEW

File: backend/Ingestion/I_Ingestion.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+#I_Ingestion.py
+#----------------------------------
+# Created By: Matthew Kastl
+#----------------------------------
+"""Ingestion is responsible for inserting data into the software. This module provides
+an interface to decouple this process from the main body of code. The factory method will 
+import and execute the correct code for a given process (based on a keyword) returning the result.
+New data will be inserted into the dataframe by reference
```
Comment (lovelysandlonelys) [inline backend/Ingestion/I_Ingestion.py:9]: period at end of sentence xoxo

File: backend/Ingestion/I_Ingestion.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    @abstractmethod
+    def ingest_data(self, data: DataFrame, **kwargs) -> bool:
+        raise NotImplementedError
+    
+
+def data_ingestion_factory(data: DataFrame, key: str, **kwargs) -> bool:
+    """ Initiates a call to a class of IDataIngestion returning the result. The call is determined by a passed key, and arguments through the kwargs.
+        :param data: DataFrame - A pre initialized data frame to insert collected data into
+        :param key: str - The string key that will be used to detect the correct module.
+        :kwargs: dict - The keyword args to pas to the resulting method. Make sure this is formatted as the targeted method wants.
```
Comment (lovelysandlonelys) [inline backend/Ingestion/I_Ingestion.py:31]: 'The keyword args to pas to" -> 'The keyword args to pass to'

File: backend/Ingestion/I_Ingestion.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        :returns bool - True indicated the process succeeded while false indicated it failed for some reason.
```
Comment (lovelysandlonelys) [inline backend/Ingestion/I_Ingestion.py:32]: weird that this sentence is in past tense and the rest aren't??

'True indicated the process succeeded while false indicated it failed for some reason.' -> 'True indicates that the process succeeded while false indicates that it failed for some reason.'

File: backend/PostProcessing/IPostProcessing.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+#IPostProcessing.py
+#----------------------------------
+# Created By: Matthew Kastl
+#----------------------------------
+"""Post processing is responsible for preforming operation on data within the collection of data.
```
Comment (lovelysandlonelys) [inline backend/PostProcessing/IPostProcessing.py:6]: operation -> operations? 
the collection -> a collection?

File: backend/PostProcessing/IPostProcessing.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+#IPostProcessing.py
+#----------------------------------
+# Created By: Matthew Kastl
+#----------------------------------
+"""Post processing is responsible for preforming operation on data within the collection of data. 
+An example of this could be preforming an arithmetic operation on a column of data or cleaning missing values.)
```
Comment (lovelysandlonelys) [inline backend/PostProcessing/IPostProcessing.py:7]: You have a close parentheses here but no open parentheses anywhere.

File: backend/PostProcessing/IPostProcessing.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+This module provides an interface to decouple this process from the main body of code. The factory method will 
+import and execute the correct code for a given process (based on a keyword) returning the result.
+New data will be inserted into the DataFrame by reference
```
Comment (lovelysandlonelys) [inline backend/PostProcessing/IPostProcessing.py:10]: what if there was a period at the end of this sentence 👀

File: backend/PostProcessing/IPostProcessing.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+class IPostProcessing(ABC):
+
+    @abstractmethod
+    def post_process(self, data: DataFrame, **kwargs) -> bool:
+        raise NotImplementedError
+    
+
+def post_process_factory(data: DataFrame, key: str, **kwargs) -> bool:
+    """ Initiates a call to a class of IDataIngestion returning the result. The call is determined by a passed key, and arguments through the kwargs.
```
Comment (lovelysandlonelys) [inline backend/PostProcessing/IPostProcessing.py:29]: Initiates a call to a class of IDataIngestion-> Initiates a call to a class of IPostProcessing?

**Review by lovelysandlonelys — APPROVED** [review]
very small documentation changes! otherwise looks good to meeee

File: data/cspec/test_1-0-0.json [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+            "48hr_VirginiaKey_wl",
+            "72hr_VirginiaKey_wl",
+            "96hr_VirginiaKey_wl"           
+          ] 
+        }
+      }
+    ],
+    "post_processing" : [],
+    "csv_config": {
+      "csv_name": "test_1-0-0.csv",
```
Comment (abeasley1722) [inline data/cspec/test_1-0-0.json:32]: Great work as always! I sent a message about this in discord.

File: backend/Ingestion/I_Ingestion.py (hunk 1 above)
Comment (matdenkas) [reply backend/Ingestion/I_Ingestion.py:9]: Got it

File: backend/Ingestion/I_Ingestion.py (hunk 2 above)
Comment (matdenkas) [reply backend/Ingestion/I_Ingestion.py:31]: Fixed

File: backend/PostProcessing/IPostProcessing.py (hunk 6 above)
Comment (matdenkas) [reply backend/PostProcessing/IPostProcessing.py:10]: What if indeed!

File: backend/PostProcessing/IPostProcessing.py (hunk 7 above)
Comment (matdenkas) [reply backend/PostProcessing/IPostProcessing.py:29]: 😅

**Review by abeasley1722 — APPROVED** [review]

**Pull request merged**